### PR TITLE
Add Spanner DB, Kubernetes Cluster and AppEngine assets

### PR DESCRIPTION
**Pull request by yardenvan**

### CODE REVIEW

File: src/main/java/com/google/cloudassets/discovery/AssetJsonParser.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * The AssetJsonParser class maps a json node into a list of Maps in which each map represents
+ * a different asset object data.
+ */
+public class AssetJsonParser {
```
Comment (itaitz) [inline src/main/java/com/google/cloudassets/discovery/AssetJsonParser.java:15]: I don't understand what the code in this file does...
Can you improve the doc, explain the response to which type of requests this parses, or even include a (possibly shortened) example of the input and output?

Comment (yardenvan) [reply src/main/java/com/google/cloudassets/discovery/AssetJsonParser.java:15]: Added more detailed documentation for this class - please let me know if it is still not clear enough

File: src/main/java/com/google/cloudassets/discovery/AssetJsonParser.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+     *
+     *  would be parsed into a List containing two maps, each with one key named "name" and its value.
+     * @param jsonNode - a JsonNode object which returns from a HTTP GET request and which contains
+     *                 the properties of a given asset object.
+     * @param assetKind - an AssetKind enum which represents for which asset kind the provided
+     *                  properties belong.
+     */
+    public AssetJsonParser(JsonNode jsonNode, AssetKind assetKind) {
+        this.assetsList = new ArrayList<>();
+        this.propertiesMap = jsonMapper.convertValue(jsonNode, Map.class);
```
Comment (itaitz) [inline src/main/java/com/google/cloudassets/discovery/AssetJsonParser.java]: Can 'propertiesMap' be a local var?

**Review by itaitz — CHANGES_REQUESTED** [review]

File: src/main/java/com/google/cloudassets/discovery/AssetJsonParser.java (hunk 2 above)
Comment (yardenvan) [reply src/main/java/com/google/cloudassets/discovery/AssetJsonParser.java]: You are right, changed it

**Review by itaitz — APPROVED** [review]

**Pull request merged**